Sources/HKStream/HKStreamRecorder.swift

```diff
@@ -109,21 +130,44 @@ public actor HKStreamRecorder {
     }
 
     /// Starts recording.
-    public func startRecording(_ fileName: String? = nil, settings: [AVMediaType: [String: any Sendable]] = HKStreamRecorder.defaultSettings) async throws {
+    /// - Parameters:
+    ///   - file: The file path for recording. If nil is specified, a unique file path will be returned automatically.
+    ///   - settings: Settings for recording.
+    public func startRecording(_ file: URL? = nil, settings: [AVMediaType: [String: any Sendable]] = HKStreamRecorder.defaultSettings) async throws {
         guard !isRecording else {
             throw Error.invalidState
         }
 
-        self.fileName = fileName ?? UUID().uuidString
-        self.settings = settings
+        var outputURL: URL
+        if let file {
```

Comment:
If you pass an URL that represents a file without a path extension (e.g. `file:///var/mobile/Containers/Data/Application/7910F2B5-9C52-42CE-87D0-20202839EB58/Documents/Demo` ) this logic would result in something like this: `file:///var/mobile/Containers/Data/Application/7910F2B5-9C52-42CE-87D0-20202839EB58/Documents/file:///var/mobile/Containers/Data/Application/7910F2B5-9C52-42CE-87D0-20202839EB58/Documents/Demo.mp4`, which is an incorrect file path.

A more future-proof suggestion to the file parameter and the outputURL handling:
  ```
      if let file {
            if file.hasDirectoryPath {
                outputURL = file.appendingPathComponent(UUID().uuidString)
            } else {
                outputURL = file
            }
            if outputURL.pathExtension == "" {
                outputURL = outputURL.appendingPathExtension(Self.defaultPathExtension)
            }
        } else {
            outputURL = moviesDirectory.appendingPathComponent(UUID().uuidString).appendingPathExtension(Self.defaultPathExtension)
        }

```

Tested with:

- URL representing a directory.
- URL representing a file without extension
- URL representing a file with extension
- nil URL

Reply:
Thank you for testing it. I’ve also added some test cases on my end. I’m thinking of adding the documentation and then merging.